src/main/java/com/example/music/backend/video/service/YoutubeVideoServiceImpl.java

```diff
@@ -0,0 +1,136 @@
+package com.example.music.backend.video.service;
+
+import com.example.music.backend.video.converter.VideoYoutubeVideoResponseMapper;
+import com.example.music.backend.video.response.YoutubeVideoResponse;
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpRequestInitializer;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.youtube.YouTube;
+import com.google.api.services.youtube.YouTubeRequestInitializer;
+import com.google.api.services.youtube.model.SearchListResponse;
+import com.google.api.services.youtube.model.SearchResult;
+import com.google.api.services.youtube.model.Video;
+import com.google.api.services.youtube.model.VideoListResponse;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.PostConstruct;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@RequiredArgsConstructor
+@Service
+public class YoutubeVideoServiceImpl implements YoutubeVideoService {
+
+    @Value("${youtube.key}")
+    private String YOUTUBE_API_KEY;
+
+    private static final HttpTransport HTTP_TRANSPORT = new NetHttpTransport();
+    private static final JsonFactory JSON_FACTORY = new JacksonFactory();
+
+    private YouTube youtube;
+
+    @PostConstruct
+    private void initializationYoutube() {
+        this.youtube = new YouTube.Builder(HTTP_TRANSPORT, JSON_FACTORY, new HttpRequestInitializer() {
+            public void initialize(HttpRequest request) {
+            }
+        }).setApplicationName("YoutubeVideo")
+                .setYouTubeRequestInitializer(new YouTubeRequestInitializer(YOUTUBE_API_KEY)).build();
+    }
+
+    private List<YoutubeVideoResponse> fromYoutubeSearchFormatToResponse(List<SearchResult> searchResults) throws IOException {
```

Comment:
Replace this method under this invocation, please

Reply:
I don't know how to combine it, because SeacrhVideo and Video are different classes, and generics won't help